src/main/java/org/truffleruby/language/loader/FeatureLoader.java

```diff
@@ -238,53 +238,65 @@ public void ensureCExtImplementationLoaded(String feature, RequireNode requireNo
                 throw new RaiseException(context.getCoreExceptions().loadError("Sulong is required to support C extensions, and it doesn't appear to be available", feature, null));
             }
 
-            final String cextRBpath = context.getRubyHome() + "/lib/truffle/truffle/cext.rb";
-            requireNode.executeRequire(cextRBpath);
+            requireNode.executeRequire(context.getRubyHome() + "/lib/truffle/truffle/cext.rb");
 
-            String rubySUpath = loadCExtLibRuby(feature);
+            final String rubySUpath = context.getRubyHome() + "/lib/cext/ruby.su";
+            final List<TruffleObject> libraries = loadCExtLibRuby(rubySUpath, feature);
 
-            sulongLoadLibraryFunction = getCExtFunction("rb_tr_load_library", rubySUpath);
+            sulongLoadLibraryFunction = RequireNode.create().findFunctionInLibraries(libraries, "rb_tr_load_library", rubySUpath);
```

Comment:
Why not reuse `requireNode` here but create a new node?

Reply:
Fixing.